WMIAdventure/frontend/src/js/components/battle/organisms/BattleView/BattleView.js

```diff
@@ -512,10 +497,39 @@ class BattleView extends React.Component {
                 enemyCompactCardTranslateY: '0'
             });
         }
-        this.callNextCardSequence();
+    }
+
+    getCurrentFullCard = () => {
+        const card = this.state.battle.getCardOnTop();
+
+        // I had to do it this way because of the way props are in FullCardView
+        if (card.level === 1) {
+            return (
+                <FullCardView cardName={card.name} cardSubject={card.subject}
+                              cardImage={card.image} cardTooltip={card.tooltip}
+                              description={card.description} common
+                              setTranslateY={this.state.fullCardAction.translateY}/>
+            )
+        } else if (card.level === 2) {
+            return (
+                <FullCardView cardName={card.name} cardSubject={card.subject}
+                              cardImage={card.image} cardTooltip={card.tooltip}
+                              description={card.description} gold
+                              setTranslateY={this.state.fullCardAction.translateY}/>
+            )
+        } else {
+            return (
+                <FullCardView cardName={card.name} cardSubject={card.subject}
+                              cardImage={card.image} cardTooltip={card.tooltip}
+                              description={card.description} epic
+                              setTranslateY={this.state.fullCardAction.translateY}/>
+            )
+        }
+
     }

```

Comment:
Jeśli Ci się chciało to chyba można zrobić osobną funkcje, co zwraca odpoweidni obiekt na podstawie levela np. `{ common: true, gold: false, epic: false}` no i potem zrobić jeden tylko `FullCardView`  zamiast trzech i mu tam podawać te wyliczone booleany